sdk/src/Services/DynamoDBv2/Custom/DataModel/InternalModel.cs

```diff
@@ -408,23 +408,51 @@ private static Dictionary<string, MemberInfo> GetMembersDictionary([DynamicallyA
         }
 
 
-
         // constructor
         internal StorageConfig([DynamicallyAccessedMembers(InternalConstants.DataModelModeledType)] Type targetType)
         {
-            if (!Utils.CanInstantiate(targetType))
+            bool isAotRuntime = InternalSDKUtils.IsRunningNativeAot();
+            bool isValid;
+            Type specificFailedType = null;
+            
+            if (isAotRuntime)
+            {
+                // In AOT mode, just do the comprehensive deep check that includes all validations
+                isValid = Utils.CanInstantiateDeep(targetType, out specificFailedType);
```

Comment:
i made a completely separate function canInstantiateDeep because i didnt want to break things with the existing CanInstantiate (although i did refactor some of the CanInstantiate into multiple re-usable methods). i could just add onto the existing method though. let me know what we think